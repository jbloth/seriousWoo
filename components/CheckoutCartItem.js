import { colors, breakPoints } from '../styles/theme';

const CheckoutCartItem = ({ product }) => {
  const { image, name, price, qty, totalPrice, size } = product; // TODO: size
  const imgUrl = image.sourceUrl;

  return (
    <div className="cart-item">
      <div
        className="img"
        style={{
          backgroundImage: `url('${imgUrl}')`,
        }}
      ></div>

      <div className="info">
        <div className="row">
          <div className="description">
            <span className="name">{name}</span>
            {size && <span className="size">{size.name}</span>}
          </div>
        </div>
        <div className="price-info">
          <span className="price--single">{price.toFixed(2)}</span>
          <span className="quantity">{`qty: ${qty}`}</span>
          <span className="price--total">{totalPrice} $</span>
        </div>
      </div>

      <style jsx>{`
        .cart-item {
          width: 100%;
          display: flex;
          border-bottom: 1px solid ${colors.violet};
          padding: 1.5rem 0;
        }

        .img {
          width: 80px;
          height: 90px;
          background-size: cover;
          background-position: center;
        }

        .info {
          flex-grow: 1;
          padding-left: 2rem;

          display: flex;
          flex-direction: column;
          justify-content: space-between;
        }

        .row {
          display: flex;
          justify-content: space-between;
        }

        .name {
          font-size: 2rem;
        }

        .size {
          margin-left: 2rem;
        }

        .price-info {
          display: flex;
          justify-content: space-between;
          align-items: center;
        }

        .price--total {
          font-size: 1.8rem;
          font-weight: bold;
        }
      `}</style>
    </div>
  );
};

export default CheckoutCartItem;
