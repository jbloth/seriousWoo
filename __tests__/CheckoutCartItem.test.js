import CheckoutCartItem from '../components/CheckoutCartItem';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import toJSON from 'enzyme-to-json';

configure({ adapter: new Adapter() });

const fakeProduct = {
  image: { sourceUrl: '/images/salmon.png' },
  name: 'Nice Shirt',
  price: '16.90€',
  qty: 2,
  totalPrice: '33.80€',
  size: { name: 'M' },
};

describe('<CheckoutCartItem/>', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<CheckoutCartItem product={fakeProduct} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const wrapper = shallow(<CheckoutCartItem product={fakeProduct} />);
    const img = wrapper.find('.img');
    const name = wrapper.find('.name');
    const size = wrapper.find('.size');
    const priceInfo = wrapper.find('.price-info');
    const price = wrapper.find('.price--single');
    const qty = wrapper.find('.quantity');
    const total = wrapper.find('.price--total');
    expect(img.props().style.backgroundImage).toBe(`url('${fakeProduct.image.sourceUrl}')`);
    expect(name.text()).toBe(fakeProduct.name);
    expect(size.text()).toBe(fakeProduct.size.name);
    expect(priceInfo.children()).toHaveLength(3);
    expect(price.text()).toBe(fakeProduct.price);
    expect(qty.text()).toBe(`qty: ${fakeProduct.qty}`);
    expect(total.text()).toBe(fakeProduct.totalPrice);
  });
});
