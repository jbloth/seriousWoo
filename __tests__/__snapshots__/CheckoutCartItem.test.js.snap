// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckoutCartItem/> matches snapshot 1`] = `
<div
  className="jsx-3212271742 cart-item"
>
  <div
    className="jsx-3212271742 img"
    style={
      Object {
        "backgroundImage": "url('/images/salmon.png')",
      }
    }
  />
  <div
    className="jsx-3212271742 info"
  >
    <div
      className="jsx-3212271742 row"
    >
      <div
        className="jsx-3212271742 description"
      >
        <span
          className="jsx-3212271742 name"
        >
          Nice Shirt
        </span>
        <span
          className="jsx-3212271742 size"
        >
          M
        </span>
      </div>
    </div>
    <div
      className="jsx-3212271742 price-info"
    >
      <span
        className="jsx-3212271742 price--single"
      >
        16.90€
      </span>
      <span
        className="jsx-3212271742 quantity"
      >
        qty: 2
      </span>
      <span
        className="jsx-3212271742 price--total"
      >
        33.80€
      </span>
    </div>
  </div>
  <JSXStyle
    dynamic={
      Array [
        "213, 182, 228",
        "23.75em",
      ]
    }
    id="1944029843"
  >
    .cart-item.__jsx-style-dynamic-selector{width:100%;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;border-bottom:1px solid rgb(213, 182, 228);padding:1.5rem 0;}.img.__jsx-style-dynamic-selector{width:80px;height:90px;background-size:cover;background-position:center;}.info.__jsx-style-dynamic-selector{-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding-left:2rem;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}.row.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;}.name.__jsx-style-dynamic-selector{font-size:2rem;}.size.__jsx-style-dynamic-selector{margin-left:2rem;}.price-info.__jsx-style-dynamic-selector{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:justify;-webkit-justify-content:space-between;-ms-flex-pack:justify;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}.price--total.__jsx-style-dynamic-selector{font-size:1.8rem;font-weight:bold;}@media only screen and (max-width:23.75em){.info.__jsx-style-dynamic-selector{padding-left:1rem;}.price-info.__jsx-style-dynamic-selector{font-size:1.4rem;}.price--total.__jsx-style-dynamic-selector{font-size:1.4rem;}}
  </JSXStyle>
</div>
`;
